change the behaviour of exported links to match the RBS and RBA documentations

doudou opened the pull request:
The velocities and accelerations were always output in the body frame, while the pose is output w.r.t. the expected target frame. Fix this to express everything in the target frame.

Hide the new behavior under the `use_proper_reference_frames` property, to avoid breaking systems that expect the old behavior.

Tests updated in https://github.com/rock-gazebo/simulation-rock_gazebo/pull/21

doudou committed:
fix reference frame in exported velocities and accelerations

The specification of the target frame, when the link is exported,
did not have any effect on the acceleration/velocity frame. This
is obviously very inconsistent.

However, this is a VERY breaking change. It's hidden behind a flag
(link_export_use_proper_references) which is currently false.

doudou committed:
hide the new reference frame behavior under a flag

This is too breaking a change to be left alone like this.

doudou commented:
@alexfneves could you check this out ?

joaobrittoneto commented:
Shouldn't the exported links' RBS follow the same pattern as the output model's RBS?
Shouldn't the definition in RBS's documentation be followed? (Maybe this is another discussion)

doudou commented:
> Shouldn't the exported links' RBS follow the same pattern as the output model's RBS?

They should match for sure. I (very wrongly) assumed that the model would be exported using the same codepath.
. 
> Shouldn't the definition in RBS's documentation be followed? (Maybe this is another discussion)

I actually should have checked that first ... This at least explains the current behavior.

@alexfneves .... sorry for the mixup, but I think what @joaobrittoneto says makes sense (the RBS definition does not, but that's a wee bit to late to change).

joaobrittoneto commented:
RBS definitions are not really clear.
Just a recap of previous discussions regarding RBS representation. 
https://github.com/rock-core/base-types/issues/99
https://github.com/rock-core/base-types/pull/100

doudou committed:
remove use_proper_reference_frames to match the RBS and RBA definitions

The documentation is clear (even if confusing), so make sure that
we're following that unconditionally

doudou commented:
@alexfneves updated to follow @joaobrittoneto remarks

doudou commented:
Closed in favor of https://github.com/rock-gazebo/simulation-orogen-rock_gazebo/pull/16
